contracts/token/ERC20/utils/SafeERC20.sol

```diff
@@ -142,14 +140,24 @@ library SafeERC20 {
      * on the return value: the return value is optional (but if data is returned, it must not be false).
      * @param token The token targeted by the call.
      * @param data The call data (encoded using abi.encode or one of its variants).
+     *
+     * This is a variant of {_callOptionalReturnBool} that reverts if call fails to meet the requirements.
      */
     function _callOptionalReturn(IERC20 token, bytes memory data) private {
-        // We need to perform a low level call here, to bypass Solidity's return data size checking mechanism, since
-        // we're implementing it ourselves. We use {Address-functionCall} to perform this call, which verifies that
-        // the target address contains contract code and also asserts for success in the low-level call.
+        uint256 returnSize;
+        uint256 returnValue;
+        assembly ("memory-safe") {
```

Comment:
I'm realizing that after https://github.com/OpenZeppelin/openzeppelin-contracts/pull/5090/commits/2fcf803f1d2e00072d4733d9b3147e8959349e96 this is not memory safe anymore. Do we care about these cases?

Reply:
57f36a78aa290289e1b4151379f55987f7e9f317 makes it memory safe again (and we probably don't care about the extra cost if we are reverting anyway)

Reply:
In a nested call we do 😑

Reply:
only if we catch ...

Reply:
As long as it is a nested call we care. It doesn't change anything if it's caught or not

Reply:
I'm not sure I understand your concern. 

This is a revert. Unless its being catched, then that is the end of the execution, and the transaction is a faillure. That are casses that we don't tend to optimize. Also, this is not movbing the free memory ptr, so even if its catched, we don't expand the memory

Reply:
> Unless its being catched, then that is the end of the execution, and the transaction is a faillure.

No, the current execution context reverts, not the transaction. We do expand memory if the returndata is larger. Although it's not an issue an external contract can returnbomb the revert too,  just inflating its price.